default/python/AI/fleet_orders.py

```diff
@@ -115,15 +118,38 @@ def can_issue_order(self, verbose=False):
         threat = f_threat + m_threat + p_threat
         safety_factor = foAI.foAIstate.character.military_safety_factor()
         universe = fo.getUniverse()
+        if main_fleet_mission.type == MissionType.INVASION:
+            # Don't advance outside of our fleet-supply zone unless the target either has no shields at all or there
+            # is already a military fleet assigned to secure the target, and don't take final jump unless the planet is
+            # (to the AI's knowledge) down to zero shields.  Additional checks will also be done by the later
+            # generic movement code
+            invasion_planet = main_fleet_mission.target
+            inv_plt_obj = invasion_planet.get_object()
+            invasion_sys = invasion_planet.get_system()
+            supplied_systems = fo.getEmpire().fleetSupplyableSystemIDs
+            #if about to leave supply lines
+            if (self.target.id not in supplied_systems and system_id in supplied_systems):
+                if (inv_plt_obj.currentMeterValue(fo.meterType.maxShield) and
+                            invasion_sys.id not in AIstate.militarySystemIDs):
+                    if verbose:
+                        print ("AIFleetOrder.OrderMove.can_issue holding Invasion fleet %d before leaving supply "+
+                               "because target planet (%s) has nonzero max shields and there is not yet a "+
+                               "military fleet assigned to secure the target system.")%(self.fleet.id, inv_plt_obj.name)
```

Comment:
1. whitespace around ```%```
2. use ```planet``` instead of ```planet.name``` (can remove "planet" in string then)
3. No need for ```+``` as strings are automatically concatenated
4. Add (): ```AIFleetOrder.OrderMove.can_issue()```